cranelift/filetests/src/test_compile.rs

```diff
@@ -63,15 +66,128 @@ impl SubTest for TestCompile {
-    let actual = text.lines().collect::<Vec<_>>();
+fn map_caperr(err: capstone::Error) -> anyhow::Error {
+    anyhow::format_err!("{}", err)
+}
+
+fn get_disassembler(isa: &dyn isa::TargetIsa) -> Result<Capstone> {
+    let cs = match isa.triple().architecture {
+        Architecture::X86_32(_) => Capstone::new()
+            .x86()
+            .mode(arch::x86::ArchMode::Mode32)
+            .build()
+            .map_err(map_caperr)?,
+        Architecture::X86_64 => Capstone::new()
+            .x86()
+            .mode(arch::x86::ArchMode::Mode64)
+            .build()
+            .map_err(map_caperr)?,
+        Architecture::Arm(arm) => {
+            if arm.is_thumb() {
+                Capstone::new()
+                    .arm()
+                    .mode(arch::arm::ArchMode::Thumb)
+                    .build()
+                    .map_err(map_caperr)?
+            } else {
+                Capstone::new()
+                    .arm()
+                    .mode(arch::arm::ArchMode::Arm)
+                    .build()
+                    .map_err(map_caperr)?
+            }
+        }
+        Architecture::Aarch64 { .. } => {
+            let mut cs = Capstone::new()
+                .arm64()
+                .mode(arch::arm64::ArchMode::Arm)
+                .build()
+                .map_err(map_caperr)?;
+            // AArch64 uses inline constants rather than a separate constant pool right now.
+            // Without this option, Capstone will stop disassembling as soon as it sees
+            // an inline constant that is not also a valid instruction. With this option,
+            // Capstone will print a `.byte` directive with the bytes of the inline constant
+            // and continue to the next instruction.
+            cs.set_skipdata(true).map_err(map_caperr)?;
+            cs
+        }
+        Architecture::S390x { .. } => Capstone::new()
+            .sysz()
+            .mode(arch::sysz::ArchMode::Default)
+            .build()
+            .map_err(map_caperr)?,
+        Architecture::Riscv64 { .. } => {
+            let mut cs = Capstone::new()
+                .riscv()
+                .mode(arch::riscv::ArchMode::RiscV64)
+                .build()
+                .map_err(map_caperr)?;
+            // Similar to AArch64, RISC-V uses inline constants rather than a separate
+            // constant pool. We want to skip dissasembly over inline constants instead
+            // of stopping on invalid bytes.
+            cs.set_skipdata(true).map_err(map_caperr)?;
+            cs
+        }
+        _ => anyhow::bail!("Unknown ISA"),
+    };
+
+    Ok(cs)
+}
+
+pub fn disassemble(isa: &dyn isa::TargetIsa, mem: &[u8]) -> Result<String> {
+    use std::fmt::Write;
+
+    let mut buf = String::new();
+
+    let cs = get_disassembler(isa)?;
+
+    let insns = cs.disasm_all(&mem, 0x0).unwrap();
+    for i in insns.iter() {
+        write!(&mut buf, "{:4x}:\t", i.address()).unwrap();
```

Comment:
Perhaps we should not display the instruction address here? Inserting or deleting an instruction will renumber all the later instructions, causing more diff churn in filetests later.

Also, it's a small thing but I'd prefer to replace all these calls to `.unwrap()` with `?`. Apparently that's straightforward since there's one `write!` invocation in this function already using `?`, so I expect it should be fine for the other writes too.

I hope it's also easy for the call to `disasm_all`, where it would be particularly good to handle errors gracefully since we might very well have generated invalid instructions. I haven't checked if the error type returned from that function converts to an `anyhow::Error` though.

Reply:
I like the output without the bytes, and thanks for catching the `unwrap()` uses! (I had pulled this directly over from clif-util, and hadn't taken the time to refactor it).

Reply:
I like the address for making jumps easier to understand, but you're right that it'll likely cause a lot of churn when updating tests. Maybe this is a good case for using `clif-util compile` when debugging, vs making the `precise-output` easier to read?